packages/api-explorer/src/components/SideNav/SideNavMethods.tsx

```diff
@@ -54,9 +54,9 @@ export const SideNavMethods = styled(
       const _isOpen = !isOpen
       setIsOpen(_isOpen)
       if (_isOpen) {
-        navigate(`/${specKey}/methods/${tag}`)
+        navigate(`/${specKey}/methods/${tag}`, { opts: null })
```

Comment:
The scenario to consider here (and with the same change in `SideNavTypes`: we are in a diff scene with selected options, and `opts` param in the url. When we open the sideNav and open a method/tag scene, this opts parameter should no longer persist. The reason we don't use `navigateWithGlobalParams` to remove this parameter is because if we started at a tag scene with a filter selected and want to select another tag, this tag should still persist when we open the new tag scene.

So, technically, the purpose of this line in plain english is more like "navigate to the tag scene and ONLY keep the tag filter variable if we have one in the URL" 

This makes me reconsider what we initially had proposed where we have a specific `navigateTo` methods. I think in the long term there will be 2 cases:

1. We navigate to a scene, remove all scene-specific params first, then push global params (currently the case with `navigateWithGlobalParams`, etc)
2. We navigate to a scene, keeping ONLY specific parameters, and removing anything else in the URL. 

To better account for case 2, I think having custom navigate methods makes the most sense. Here we'd explicitly have a `navigateToTagScene` instead of having to explicitly nullify any extra parameters that could persist when navigating to a tag scene from the SideNav.

Reply:
@josephaxisa